cpp/src/rolling/rolling.cu

```diff
@@ -163,7 +163,7 @@ std::enable_if_t<!std::is_same<InputType, cudf::string_view>::value and
   bool output_is_valid = (count >= min_periods);
 
   // store the output value, one per thread
-  cudf::detail::store_output_functor<OutputType, op == aggregation::MEAN>{}(
+  cudf::detail::rolling_store_output_functor<OutputType, op == aggregation::MEAN>{}(
```

Comment:
I don't understand the changes to this file  -- they seem to have nothing to do with list_view?

Reply:
Agreed, looks like a bad merge?

Reply:
Right, so this requires some explanation.  This was done in support of list_view SFINAE.  There were several of issues:

- The namespace is just cudf::detail, so it was very odd to have traits called `is_supported`,  `is_string_supported`, etc.  That could mean anything, so I added the "rolling" prefix to clarify things.

- There was an unused template parameter being passed around needlessly.  op == aggregation::MEAN>() which cluttered everything up.

- The semantics of `is_supported` and `is_string_supported` were not what you'd expect.  As an example COUNT is indeed valid for strings.  But is_supported<string, COUNT> would return false.   The logic in the original version of `is_supported` was also pretty hard to read.  So I changed the semantics to be as follows: 

   `is_supported` becomes `is_rolling_supported` and simply means "yes, this operation is
   supported on this type, period". So, `is_rolling_supported<string, COUNT>` returns true.

   `is_string_supported` becomes `is_string_specialized` and means "this is the specialization
   that implements the support for strings"

   Put together, this has several effects.  First, it just makes the semantics more clear in the
   SFINAE.

```
     // this operation is supported and not specialized for strings
     std::enable_if_t<cudf::detail::is_rolling_supported<T, agg_op, op>() and
                      !cudf::detail::is_rolling_string_specialized<T, agg_op, op>()>

     vs

     // this operation is supported but not supported for strings.  
     // (??? even though it -is- supported for strings, just not right here)
     // The SFINAE for the string implementation uses 
     // (!is_supported and is_string_supported) which is also odd.
     std::enable_if_t<(cudf::detail::is_supported<T, agg_op, op>()) and
                              !(cudf::detail::is_string_supported<T, agg_op, op>())>
```
   
   Second, it considerably simplifies the logic in `is_supported` / `is_rolling_supported` and 
   makes it easier to add new types (eg list).  All you have to do is add the logic for whether 
   the operation is supported without any concern for other conditions.

   Third, it allowed COUNT to not require code duplication for strings.  Previously, the launch() for MIN, MAX, COUNT_VALID and COUNT_ALL didn't include string because MIN/MAX needed specialization.   But because of this , COUNT_VALID and COUNT_ALL had to be added to the string specialized launch().   With this new setup, it doesn't have to. 

   Finally, and most relevant, it allows things that aren't supported to trivially fall through to the "invalid combination" SFINAE template.  That's why you don't see any list_view code in this file.  The logic in is_rolling_supported() sends list_view to COUNT only with no other changes needed.  If we want to add support for lists for a specific aggregation, simply add is_list_specialized().   Without this rework, there would be a considerably amount of complexity added to the SFINAE logic to get list_view to fall through like that.

Reply:
> `is_string_supported` becomes `is_string_specialized` and means "this is the specialization that implements the support for strings"

Ok, I do prefer the new function names and semantics.

It turns out that `cudf::detail::is_supported()` wasn't using its `is_mean` parameter at all. Thanks for cleaning that up.

One minor suggestion is to rename `is_rolling_string_specialized<>` to `is_rolling_string_specialization<>`, to indicate that *this* is the SFINAE specialization of interest.

Reply:
Agree. Change made.